File: PharmaDicBackEnd/PharmaDicBackEnd.ApiService/Models/Ingredient.cs
using System;
using System.Collections.Generic;

namespace PharmaDicBackEnd.ApiService.Models;

public partial class Ingredient
{
    public int IngredientId { get; set; }

    public string IngredientName { get; set; } = null!;

    public string? Description { get; set; }

    public virtual ICollection<MedicineIngredient> MedicineIngredients { get; set; } = new List<MedicineIngredient>();
}

File: PharmaDicMobile/app/src/main/java/com/example/pharmadicmobile/models/Ingredient.java
package com.example.pharmadicmobile.models;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Ingredient implements Serializable {
    private int ingredientId;
    private String ingredientName;
    private String description;
    private List<MedicineIngredient> medicineIngredients = new ArrayList<>();

    public Ingredient() {}

    public int getIngredientId() { return ingredientId; }
    public void setIngredientId(int ingredientId) { this.ingredientId = ingredientId; }

    public String getIngredientName() { return ingredientName; }
    public void setIngredientName(String ingredientName) { this.ingredientName = ingredientName; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public List<MedicineIngredient> getMedicineIngredients() { return medicineIngredients; }
    public void setMedicineIngredients(List<MedicineIngredient> medicineIngredients) { this.medicineIngredients = medicineIngredients; }
}
